### ISSUE TITLE
Create an `equals` function to compare different things

### ISSUE THREAD

**Issue by joente** [thingsdb/ThingsDB#144]
For things to be considered equal in ThingsDB, they must be truly equal.

For example: 
```
assert( {name: 'test'} != {name: 'test'} );
```

Sometimes however, it would be useful to perform a weak compare where things are only compared by their keys and values, for example:
```
assert( {name: 'test'}.equals({name: 'test'}) );
```

Because an unlimited *deep* compare might result in an endless loop, we can perform the compare only 1 level deep. To solve this issue of comparing an object several layers deep, the `equals` function could accept an optional "deep" argument.

**Comment by joente** [comment]
- [x] Test code
- [x] Documentation
- [x] Code
- [x] Syntax Highlighting
  - [x] VSCode
  - [x] Highlight.js
  - [x] Create issue for ThingsGUI

**Comment by joente** [comment]
Implemented in **v0.10.2-alpha-7**
